src/TextInput.js

```diff
@@ -1,167 +1,145 @@
-import React, { Component } from 'react';
+import React, { useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import idgen from './idgen';
 import constants from './constants';
 
-class TextInput extends Component {
-  constructor(props) {
-    super(props);
-
-    this.id = props.id || idgen();
-
-    if (props.password) {
-      this.id = `password${idgen()}`;
-    }
-
-    if (props.email) {
-      this.id = `email${idgen()}`;
-    }
+const TextInput = props => {
+  const id = props.id || idgen();
```

Comment:
I'm going to merge this since the task in hand is 99% done, although here we have a potential bug, every time this component rerenders it will create another `id` and this is not desired, if you want to overwrite the id prop it should live in `defaultProps` I will create a follow up PR and link it here